ml-agents/mlagents/trainers/subprocess_env_manager.py

```diff
@@ -180,6 +180,8 @@ def _generate_all_results() -> AllStepResult:
         UnityTimeOutException,
         UnityEnvironmentException,
         UnityCommunicatorStoppedException,
+        # NOTE - any other exception will cause a deadlock
+        AssertionError,
```

Comment:
See https://jira.unity3d.com/browse/MLA-1199 - this is being fixed separately.

I'll revert this line if all other tests pass.